src/js/hopscotch.js

```diff
@@ -1,4 +1,22 @@
-(function(context, namespace) {
+(function(context, factory) {
+  'use strict';
+
+  if (typeof define === 'function' && define.amd) {
+    // AMD. Register as an anonymous module.
+    define([], factory);
+  } else if (typeof exports === 'object') {
+    // Node/CommonJS
+    module.exports = factory();
+  } else {
+    var namespace = 'hopscotch';
+    // Browser globals
+    if (context[namespace]) {
+      // Hopscotch already exists.
+      return;
+    }
+    context[namespace] = factory(context.jQuery, context.YAHOO);
+  }
+}(this, (function(jQuery, YAHOO) {
```

Comment:
We should remove jQuery and YAHOO from factory params, since they are optional library dependencies. Somebody quickly looking at the source code might assume they are hard dependencies. You should still be able to do `hasJquery         = (typeof jQuery !== undefinedStr),` inside factory method, since it will assume window scope anyway.